docs/concepts/whatisvpp/modularflexibleextensible.rst

```diff
@@ -0,0 +1,43 @@
+.. _modularflexibleextensible:
+
+.. toctree::
+
+Modular, Flexible, and Extensible
+***********************************
+
+The FD.io VPP platform is built on a ‘packet processing graph’. This modular
+approach means that anyone can ‘plugin’ new graph nodes. This makes
+extensibility rather simple, and it means that plugins can be customized
+for specific purposes.
+
+.. figure:: /_images/VPP_custom_application_packet_processing_graph.280.jpg
+   :alt: Custom Packet Processing Graph
+
+   Custom Packet Processing Graph
+
+How does the plugin come into play? At runtime, the FD.io VPP platform grabs
+all available packets from RX rings to form a vector of packets. A
+packet processing graph is applied, node by node (including plugins) to
```

Comment:
The received packet's typically traverse the packet processing graph nodes in the vector, when the network processing represented by each graph node is applied to each packet in turn.